### DISCUSSION TITLE
想问下pnpm缓存通过哪种方式最好 用的alinas反而没有download快

### DISCUSSION THREAD

**Discussion by chenemiyi** [pnpm/pnpm#7363] [Q&A]
Content-addressable store is at: /root/.pnpm-store/v3
 Virtual store is at: node_modules/.pnpm
 WARN  EXDEV: cross-device link not permitted, link '/root/.pnpm-store/v3/files/e2/259701f5d70901ead0eaf1b6b7606c466b167169e57d243c3865900570393aacc723ae0ab9628bed5382dc76ffd820e33df9b18ea9d2a6cc0970dda8307dd1' -> '/home/jenkins/workspace/ask-sharelib-test/node_modules/.pnpm/npm-verdaccio.gaodunwangxiao.com+follow-redirects@1.15.3_debug@4.3.4/node_modules/_tmp_123_b8f40937d44f403096c960a00724ef87/LICENSE'
Falling back to copying packages from store

node 14.21.3
pnpm 7.33.6

**Comment by RexSkz** [comment]
You can not hard-link a file from one disk to another due to how the hard link works. A good way to take advantage of the speed is to *set the `.pnpm-store` root to the disk where your project placed.*

P.S. The new copy-on-write feature is not working across devices either :)
